### PR TITLE
TM-6123, TM-6126, Re-Interate Projected Vacancy

### PR DESCRIPTION
**Pull request by M00staff**
Dual Merge: 
- [API PR](https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/1017)
- [Mock PR](https://github.com/MetaPhase-Consulting/mock-fsbid/pull/484)

[Ticket](https://metaphase.atlassian.net/browse/TM-6123)
[Ticket](https://metaphase.atlassian.net/browse/TM-6126)

### CODE REVIEW

File: src/Components/BureauPage/ProjectedVacancy/ProjectedVacancy.jsx [hunk 1]
```diff
@@ -7,8 +7,12 @@ import { checkFlag } from 'flags';
 import { sortBy } from 'lodash';
 import { usePrevious } from 'hooks';
 import {
-  projectedVacancyEdit, projectedVacancyFetchData, projectedVacancyFilters,
-  projectedVacancyLangOffsetOptions, projectedVacancyLangOffsets, saveProjectedVacancySelections,
+  projectedVacancyEdit,
+  projectedVacancyFetchData,
+  projectedVacancyFilters,
+  projectedVacancyLangOffsetOptions,
+  // projectedVacancyLangOffsets,
```
Comment (M00staff) [inline src/Components/BureauPage/ProjectedVacancy/ProjectedVacancy.jsx:14]: i guess we can remove this?

File: src/Components/BureauPage/ProjectedVacancy/ProjectedVacancy.jsx [hunk 2]
```diff
@@ -22,14 +26,14 @@ import ProfileSectionTitle from 'Components/ProfileSectionTitle/ProfileSectionTi
...
 const enableCycleImport = () => checkFlag('flags.projected_vacancy_cycle_import');
+const enableEdit = () => checkFlag('flags.projected_vacancy_edit');
 
 // eslint-disable-next-line complexity
 const ProjectedVacancy = () => {
   const dispatch = useDispatch();
 
   const userProfile = useSelector(state => state.userProfile);
   const isAo = userHasPermissions(['ao_user'], userProfile?.permission_groups);
-  const isBureau = userHasPermissions(['bureau_user'], userProfile?.permission_groups);
```
Comment (M00staff) [inline src/Components/BureauPage/ProjectedVacancy/ProjectedVacancy.jsx:32]: dropped this cause i moved the include checkbox to inside the Edit part & AO's cannot edit 🤷

File: src/Components/BureauPage/ProjectedVacancy/ProjectedVacancy.jsx [hunk 3]
```diff
@@ -54,10 +58,8 @@ const ProjectedVacancy = () => {
   const pageSizes = PANEL_MEETINGS_PAGE_SIZES;
   const count = positions?.length;
 
-  const [includedPositions, setIncludedPositions] = useState([]);
```
Comment (M00staff) [inline src/Components/BureauPage/ProjectedVacancy/ProjectedVacancy.jsx:57]: completely re-did how the include button worked - seemed weird to have them bulk call an endpoint one-by-one 

image

image

not entirely sure what it does - or why it was like that to begin with 🤷

also just used a drop down cause it was simpler to style - but ran into an issue with the fact that Y/N/Null are all options - so there's just a blank option... again 🤷

with the time constraints, lack of context and just picking this work up from Patrick - this is the best i could come up with

File: src/Components/BureauPage/ProjectedVacancy/ProjectedVacancy.jsx [hunk 4]
```diff
@@ -312,7 +254,7 @@ const ProjectedVacancy = () => {
       }
     });
     const editData = { projected_vacancy: updatedPvs };
-    dispatch(projectedVacancyEdit(getQuery(), editData));
+    dispatch(projectedVacancyEdit(getQuery(), editData)); // swapped out PV Endpoints - this no longer works
```
Comment (M00staff) [inline src/Components/BureauPage/ProjectedVacancy/ProjectedVacancy.jsx:257]: yea - also, zero idea what the Include functionality does or how it works, but it doesn't work anymore 👍

File: src/Components/ProjectedVacancyCard/ProjectedVacancyCard.jsx [hunk 5]
```diff
@@ -59,19 +52,15 @@ const ProjectedVacancyCard = (props) => {
...
-      post_bidding_considerations_url: result?.obc_url,
-    },
-  };
+  // const differentials = {
+  //   post: {
+  //     danger_pay: result?.bidding_tool_danger_rate_number,
+  //     differential_rate: result?.bidding_tool_differential_rate_number,
+  //     post_bidding_considerations_url: result?.obc_url,
+  //   },
+  // };
```
Comment (M00staff) [inline src/Components/ProjectedVacancyCard/ProjectedVacancyCard.jsx:63]: 🤷

File: src/Components/ProjectedVacancyCard/ProjectedVacancyCard.jsx [hunk 6]
```diff
@@ -169,29 +142,29 @@ const ProjectedVacancyCard = (props) => {
...
+      // {
+      //   'Language Offset Summer': summerLanguageOffsets?.find(o =>
+      //     o.code === languageOffsets?.language_offset_summer)?.description || DEFAULT_TEXT,
+      // },
+      // {
+      //   'Language Offset Winter': winterLanguageOffsets?.find(o =>
+      //     o.code === languageOffsets?.language_offset_winter)?.description || DEFAULT_TEXT,
+      // },
+      { 'PP/Grade': result?.combinedppgrade },
+      // { 'Post Differential | Danger Pay': getDifferentials(differentials) },
```
Comment (M00staff) [inline src/Components/ProjectedVacancyCard/ProjectedVacancyCard.jsx:165]: 🤷

i wonder if this has something to do with the error i was getting in the API around `pospayplancode` ?

**Comment by M00staff** [comment]
i gotta throw it out there again - I'm not sure its a good idea to have 3 developers simultaneously working on the same feature & code blocks.... This is getting pretty messy for my taste, and the amount of dead & disjointed code is piling up.

**Review by jlam-mpc — APPROVED** [review]

File: src/Components/BureauPage/ProjectedVacancy/ProjectedVacancy.jsx (hunk 1 above)
Comment (jlam-mpc) [reply src/Components/BureauPage/ProjectedVacancy/ProjectedVacancy.jsx:14]: Leave for now but yes code cleanup on this component would be great

**Pull request merged**